tools/Docker-dev/Dockerfile

```diff
@@ -0,0 +1,143 @@
+ARG miniforge_patch_number="5"
+# Miniforge installer architecture
+ARG miniforge_arch="x86_64"
+# Python implementation to use 
+# can be either Miniforge3 to use Python or Miniforge-pypy3 to use PyPy
+ARG miniforge_python="Miniforge3"
+
+# Miniforge archive to install
+ARG miniforge_version="${conda_version}-${miniforge_patch_number}"
+# Miniforge installer
+ARG miniforge_installer="${miniforge_python}-${miniforge_version}-Linux-${miniforge_arch}.sh"
+# Miniforge checksum
+ARG miniforge_checksum="49dddb3998550e40adc904dae55b0a2aeeb0bd9fc4306869cc4a600ec4b8b47c"
+
+# -----------------------------------------------------------------------------
+# ---- Python version to install ----
+# Currently Python 3.8
+ARG PYTHON_VERSION=default
+
+# ---- Configure environment ----
+ENV CONDA_DIR=/opt/conda \
+    SHELL=/bin/bash 
+
+ENV PATH=$CONDA_DIR/bin:$PATH \
+    CONDA_VERSION="${conda_version}" \
+    MINIFORGE_VERSION="${miniforge_version}" \
+    CONDA_ENV=scipydev \
+    PROJECT_DIR=/home/scipy
+
+# -----------------------------------------------------------------------------
+# ---- OS dependencies ----
+ENV DEBIAN_FRONTEND noninteractive
+
+# hadolint ignore=DL3008
+RUN apt-get update && \ 
+    apt-get install -yq --no-install-recommends \
+    software-properties-common \
+    python3.8-dev \
+    build-essential \
+    vim \
+    libatlas-base-dev \
+    gfortran \
+    libgfortran4 \
+    liblapack-dev \
+    curl \
+    libgmp-dev \
+    libmpfr-dev \
+    libsuitesparse-dev \
+    libmpc-dev \
+    wget \
+    ca-certificates && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+# -----------------------------------------------------------------------------
+
+# Enable prompt color in the skeleton .bashrc
+# hadolint ignore=SC2016
+RUN sed -i 's/^#force_color_prompt=yes/force_color_prompt=yes/' /etc/skel/.bashrc && \
+    # Add call to conda init script see https://stackoverflow.com/a/58081608/4413446
+    echo 'eval "$(command conda shell.bash hook 2> /dev/null)"' >> /etc/skel/.bashrc 
+
+# -----------------------------------------------------------------------------
+# ---- Installing conda  ----
+RUN wget --quiet "https://github.com/conda-forge/miniforge/releases/download/${miniforge_version}/${miniforge_installer}" && \
+    echo "${miniforge_checksum} *${miniforge_installer}" | sha256sum --check && \
+    /bin/bash "${miniforge_installer}" -f -b -p $CONDA_DIR && \
+    rm "${miniforge_installer}" && \
+    # Conda configuration see https://conda.io/projects/conda/en/latest/configuration.html
+    echo "conda ${CONDA_VERSION}" >> $CONDA_DIR/conda-meta/pinned && \
+    conda config --system --set auto_update_conda false && \
+    conda config --system --set show_channel_urls true && \
+    if [ ! $PYTHON_VERSION = 'default' ]; then conda install --yes python=$PYTHON_VERSION; fi && \
+    conda list python | grep '^python ' | tr -s ' ' | cut -d '.' -f 1,2 | sed 's/$/.*/' >> $CONDA_DIR/conda-meta/pinned && \
```

Comment:
Could you add a comment for this line? I'm not quite sure what it does.

Reply:
Will do - this allows for the Python version (currently 3.8) to be changed at build-time by providing a build arg